spec/derivativo/user_path_helper_spec.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Derivativo::UserPathHelper do
+  describe '.which' do
+    context 'in a unix-like environment' do
+      before do
+        stub_const('File::PATH_SEPARATOR', ':') # so that texts run properly regardless of OS running them
+        allow(described_class).to receive(:path_values).and_return(['/the/first/path', '/the/second/path', '/the/third/path'])
+        allow(File).to receive(:executable?).and_return(false) # default negative match case
+        allow(File).to receive(:executable?).with('/the/second/path/ffmpeg').and_return(true) # positive match case
+        allow(described_class).to receive(:case_sensitive_file_path).and_return(false) # default negative match case
+        allow(described_class).to receive(:case_sensitive_file_path).with('/the/second/path', 'ffmpeg').and_return('/the/second/path/ffmpeg') # positive match case
+      end
+      it 'finds a program on the path' do
+        expect(described_class.which('ffmpeg')).to eq('/the/second/path/ffmpeg')
+      end
+
+      it 'does not find a program that is not on the path' do
+        expect(described_class.which('not_a_real_program')).to eq(nil)
```

Comment:
Because File.executable? is stubbed out, won't hit the TypeError when calling the real File.executable? with nil as an argument.

Reply:
Fixed through my recent test improvements!  (Specifically, the case_sensitive_file_path tests.)  Thanks for catching this!